app/views/questionnaire.py

```diff
@@ -219,40 +210,125 @@ def get_session_expired(eq_id, form_type, collection_id):  # pylint: disable=unu
                                  survey_title=TemplateRenderer.safe_content(schema['title']))
 
 
+@questionnaire_blueprint.route('<group_id>/<int:group_instance>/permanent-or-family-home', methods=["POST"])
+@login_required
+def post_everyone_at_address_confirmation(eq_id, form_type, collection_id, group_id, group_instance):
+    if request.form.get('permanent-or-family-home-answer') == 'No':
+        _remove_repeating_on_household_answers(get_answer_store(current_user), group_id)
+    return post_block(eq_id, form_type, collection_id, group_id, group_instance, 'permanent-or-family-home')
+
+
+def _render_block(block, current_location, post_form=None):
+
+    if post_form is None:
+        context = _get_context(block, current_location, get_answer_store(current_user))
+    else:
+        context = {'form': post_form, 'block': block}
+
+    full_routing_path = path_finder.get_routing_path()
+
+    return _build_template(
+        current_location,
+        context,
+        block['type'],
+        routing_path=full_routing_path)
+
+
+def _valid_location(location):
+    return location in path_finder.get_routing_path(
+        location.group_id,
+        location.group_instance,
+    )
+
+
+def _valid_group(group_id):
```

Comment:
I think we should make the naming of these private methods consistent?
e.g. the ones that return a bool start with is. e.g. `_is_skipping_to_the_end`
it just makes it clearer in my opinion